akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -228,9 +228,10 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
     else {
-      val putDeltaOp = PutDeltaOp(keys.resetDelta.add(node, key).delta.get, key → value, zeroTag)
-      // put forcibly damages history, so we propagate full value that will overwrite previous values
-      new ORMap(keys.add(node, key), values.updated(key, value), zeroTag, Some(newDelta(putDeltaOp)))
+      val newKeys = keys.resetDelta.add(node, key)
```

Comment:
ah, I was starring to this and couldn't understand why this would fix it since everything is immutable. Now I understand.